quixstreams/dataframe/series.py

```diff
@@ -9,11 +9,34 @@
 from quixstreams.core.stream.stream import Stream
 from quixstreams.models.messagecontext import MessageContext
 from .base import BaseStreaming
-from .exceptions import InvalidOperation
+from .exceptions import InvalidOperation, MissingColumn, InvalidColumnReference
 
 __all__ = ("StreamingSeries",)
 
 
+def _getitem(d: Mapping, column_name: Union[str, int]) -> object:
+    """
+    Special error handling around column referencing with SDF.
+
+    :param d: a dict-like object (usually just a dict).
+    :param column_name: the column name.
+
+    :return: Nested data from column name
+    """
+    try:
+        return d[column_name]
+    except KeyError:
+        raise MissingColumn(
+            f"Column name '{column_name}' does not exist in the StreamingDataFrame"
+        )
+    except TypeError:
+        d_type = getattr(type(d), "__name__", type(d))
```

Comment:
Why do you need to use `getattr` here?

`str(type(d))` looks much shorter

Reply:
the `__name__` changes it from something like `<class 'str'>` to just `'str'`, which I thought was nicer to read (and then could not assume everything thrown at it would have `__name__` defined, so I did `getattr`)

of course, it'll treat it like `str(type(d))` otherwise =)

Let me know if you still don't like it.

Reply:
Let's  do `obj.__class__.__name__` instead